tests/test_localtime.py

```diff
@@ -1023,3 +1023,71 @@ def test_timestamp_british_summer_time_after_clocks_move_backward(self):
         assert dt.day == 25
         assert dt.hour == 1
         assert dt.minute == 30
+
+
+class TestGetMonthDatetimeRanges:
+    def test_yields_correct_ranges(self):
```

Comment:
🤔 This test is good, but I think it would be helpful to check a couple of other scenarios. Specifically:
* What happens if the input range is not aligned to month boundaries?
* What happens if the timezone is a weird one (a normal-weird example would be AUS, a super-weird one would be Samoa between 2011 and 2012)

Reply:
> Do you mean providing the timezone within the datetime objects, or as an argument for the function? 

Both, but the latter is the most important